test/test-runner/src/runner.rs

```diff
@@ -82,7 +83,9 @@ impl Runner {
     }
 
     pub fn run_case(&self, config: &suites::Config, case: &Case) -> Report {
-        let input = case.content().unwrap();
+        let canonicalized_path = case.path.clone().canonicalize().unwrap();
```

Comment:
Is `canonicalize` necessary here, i.e. resolving symlinks? Otherwise, `absolute` may be sufficient: https://doc.rust-lang.org/std/path/fn.absolute.html.
As a general rule, I would be hesitant with resolving symlinks. If you ever need to persist paths (which should of course be avoided if possible), you generally want to persist the unresolved path to avoid breakage when the symlink target changes.

Reply:
I was discouraged by the "This is a nightly-only experimental API." comment on the documentation of `absolute`.
The path contained in `case.path` itself is relative (something like "examples/foo.pol"), and I just need to do the next best thing so that `Url::from_file_path` doesn't fail. Would you say that `absolute` is better here?

Reply:
I didn't realize that `absolute` is unstable. Let's stick with `canonicalize` then.